src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -1621,5 +1627,45 @@ private static void ComputeFolderNames(uint folderItemID, List<string> names, IV
             }
         }
         #endregion
+
+        private string GetDefaultNamespace(IVsHierarchy hierarchy)
```

Comment:
return type of this needs to be documented.  it's not at all clear to me what this shoudl be returning.  impl returns any of 'null', 'empty' and an actual default namespace.